sync/src/mutex.rs

```diff
@@ -0,0 +1,347 @@
+// Copyright (c) The Rust Foundation
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! mutex.rs implementation more or less copied from
+//! [rust source](https://github.com/rust-lang/rust.git) at
+//! [606c3907](https://github.com/rust-lang/rust/commit/606c3907251397a42e23d3e60de31be9d32525d5)
+//!
+//! Differences:
+//! - The imports were changed to work with the `mc-sgx` crates.
+//! - The stable attributes have been removed
+//! - The unstable attributes have been removed
+//! - Items that are crate only were converted from `pub` to `pub(crate)`
+//! - Removed examples that were not possible in an SGX enclave have been omitted
+//! - Ran `cargo fmt`
+//! - Removed unnecessary unsafe blocks
+
+#![allow(dead_code)]
+
+use crate::sys::locks as sys;
+use crate::{poison, LockResult, TryLockError, TryLockResult};
+use core::cell::UnsafeCell;
+use core::fmt;
+use core::ops::{Deref, DerefMut};
+
+/// A mutual exclusion primitive useful for protecting shared data
+///
+/// This mutex will block threads waiting for the lock to become available. The
```

Comment:
Great description.

Reply:
I wish I could take credit, but it's from [mutex.rs](https://github.com/rust-lang/rust/blob/master/library/std/src/sync/mutex.rs)